src/Microsoft.SqlTools.ServiceLayer/DacFx/Contracts/SavePublishProfileRequest.cs

```diff
@@ -0,0 +1,54 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+using System.Collections.Generic;
+using Microsoft.SqlServer.Dac;
+using Microsoft.SqlTools.Hosting.Protocol.Contracts;
+//using Microsoft.SqlTools.ServiceLayer.Utility;
+
+namespace Microsoft.SqlTools.ServiceLayer.DacFx.Contracts
+{
+    /// <summary>
+    /// Parameters for a DacFx save publish profile request.
+    /// </summary>
+    public class SaveProfileParams
+    {
+        /// <summary>
+        /// Gets or sets the profile path
+        /// </summary>
+        public string ProfilePath { get; set; }
+
+        /// <summary>
+        /// Gets or sets name for database
+        /// </summary>
+        public string? DatabaseName { get; set; }
+
+        /// <summary>
+        /// Gets or sets target connection string
+        /// </summary>
+        public string? ConnectionString { get; set; }
+
+        /// <summary>
+        /// Gets or sets SQLCMD variables for deployment
+        /// </summary>
+        public IDictionary<string, string>? SqlCommandVariableValues { get; set; }
+
+        /// <summary>
+        /// Gets or sets the options for deployment
+        /// </summary>
+        public DeploymentOptions? DeploymentOptions { get; set; }
+    }
+
+    /// <summary>
+    /// Defines the DacFx save publish profile request type
+    /// </summary>
+    class SavePublishProfileRequest
+    {
+        public static readonly RequestType<SaveProfileParams, bool> Type =
```

Comment:
usually the return type is or a request is `ResultStatus`, rather than bool so that both success and errormessage can be passed back https://github.com/microsoft/sqltoolsservice/blob/62e896c94cf60c51f04c4b45fed5ab269d821a23/src/Microsoft.SqlTools.ServiceLayer/Utility/ResultStatus.cs

Reply:
Updated it to return ResultStatus. As of now, added a TODO tag to return something from Save method as the success/failure.